addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/provisioner-service.js

```diff
@@ -432,6 +438,25 @@ class ProvisionerService extends Service {
     providerConfig.userPoolDomain = userPoolDomain;
     return providerConfig;
   }
+
+  async retryCreateDomain(cognitoIdentityServiceProvider, params, userPoolDomain) {
+    // Cognito requires domain prefix to be 63 or shorter
+    params.Domain = generateIdSync(userPoolDomain)
```

Comment:
I guess userPoolDomain cannot be 63 characters but if it is even accidentally then it seems like an infinite recursion. Perhaps limit the retries to 10 or something by passing an additional parameter of retryCount to this function? This will be a good unit test to add too.